capycli/common/script_base.py

```diff
@@ -46,15 +46,21 @@ def login(self, token: str = "", url: str = "", oauth2: bool = False) -> bool:
             self.sw360_url = url
 
         if not self.sw360_url:
-            print_red("  No SW360 server URL specified!")
-            sys.exit(ResultCode.RESULT_ERROR_ACCESSING_SW360)
+            if exit_no_login:
+                print_red("  No SW360 server URL specified!")
+                sys.exit(ResultCode.RESULT_ERROR_ACCESSING_SW360)
```

Comment:
Note that you could also catch `SystemExit` in the caller of `login()`. Not sure if we want an extra parameter for this. What do you think, @tngraf?

Reply:
I see no reason for an extra parameter.
@sachinshaji Please explain why this should be needed.

Reply:
As we have added the functionality to lookup the SW360 server(to find the already existing source code) for the `capycli bom findsources` command we need SW360 URL and TOKEN. **If we don't add the** ` exit_no_login` paramerter to `login` the `capycli bom findosurces` will always search for SW360 URL and TOKEN and if we don't give, the command will fail/error out.

We have added the ` exit_no_login` so that even the users doesn't give SW360 creds the `capycli bom findsources` will run successfully trying to find the source code from github(only skip the SW360 lookup).

Reply:
I stll do not see the need:
* Case 1: a token and a SW360 url have been given on the command line.
  You can check that they are not emptry and then call `login()` => no `exit_no_login`needed.
* Case 2: neither token nor SW360 url have been given on the command line.
  a) The user decided not to provide this information => user does not want to use SW360 information   
     => do not call login => no `exit_no_login`needed.  
  b) You can mimik the behavior of `login()` and check if the information is available in the environment.
      If is exists, pass it as parameter. If it does not exist you cannot login.
      In either way no `exit_no_login`needed.

Reply:
Ok I will check this. Thanks for the detailed explanation.